### PR TITLE
Add ansible-navigator support

### PR DESCRIPTION
**Pull request by anxjok**
Hey, I added ansible-navigator support for molecule. 
The ansible-navigator provisioner is a Derived class from the ansible provisioner class. 
To get the stdout output from ansible-navigator I parse the json artifacts from ansible navigator.

With the implementation of ansible-navigator, the roles or playbooks can be tested against a modified version of an awx-ee image, to ensure it works with the awx-ee image that is used in AAP or AWX.

I created a sample  [repository](https://github.com/anxjok/creator-ee) to modify the creator-ee to support docker from the host inside the ee.

### CODE REVIEW

File: src/molecule/config.py [hunk 1]
```diff
@@ -221,6 +221,8 @@ def provisioner(self):
         provisioner_name = self.config["provisioner"]["name"]
         if provisioner_name == "ansible":
             return ansible.Ansible(self)
+        elif provisioner_name == "ansible-navigator":
+            return ansible_navigator.Ansible_Navigator(self)
```
Comment (zhan9san) [inline src/molecule/config.py:225]:
```suggestion
            return ansible_navigator.AnsibleNavigator(self)
```

**Review by zhan9san — CHANGES_REQUESTED** [review]

File: src/molecule/config.py (hunk 1 above)
Comment (anxjok) [reply src/molecule/config.py]: I applied these challenges in the last commit

**Comment by bbaassssiiee** [comment]
branch has conflicts that must be resolved

**Comment by ssbarnea** [comment]
I will close this because at this moment we don't want to use navigator from within molecule. Still, we will add integration but without making molecule aware about being run under navigator.

**Pull request closed without merging**